Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -0,0 +1,25 @@
+class JavaRequirement
+  cask "java"
+
+  satisfy build_env: false do
+    preferred_java_home
+    next false unless @java_home
```

Comment:
Note to self: it looks like this needs additional code to handle version-checking.